src/gluonts/shell/train.py

```diff
@@ -12,233 +12,118 @@
+            'enough data to be used for evaluation. Proceeding with '
+            f'{len_filtered}/{len_orig} '
+            f'(~{int(len_filtered/len_orig*100)}%) items.'
+        )
+    return test_dataset
 
 
-def get_estimator_path(hyperparameters: dict) -> str:
-    if 'SWIST_ESTIMATOR_CLASS' in os.environ:
-        logger.info(
+def get_forecaster(forecaster_class=None):
+    if 'GLUONTS_ESTIMATOR_CLASS' in os.environ:
+        log.logger.info(
             'Picking up GluonTS estimator classname from a '
-            '"SWIST_ESTIMATOR_CLASS" environment variable.'
+            '"GLUONTS_ESTIMATOR_CLASS" environment variable.'
         )
-        return os.environ['SWIST_ESTIMATOR_CLASS']
+        forecaster_class = os.environ['GLUONTS_ESTIMATOR_CLASS']
 
-    elif 'estimator_class' in hyperparameters:
-        logger.info(
-            'Picking up GluonTS estimator classname from an "estimator_class" '
+    elif forecaster_class is not None:
+        log.logger.info(
+            'Picking up GluonTS estimator classname from an "forecaster_class" '
             'hyperparameter value.'
         )
-        return hyperparameters['estimator_class']
-
-    raise GluonTSFatalError(
-        'Cannot determine the GluonTS estimator classname (missing variable '
-        '"SWIST_ESTIMATOR_CLASS").'
-    )
-
-
-EstimatorT = TypeVar('EstimatorT', bound=Estimator)
-
-
-def load_estimator(class_path: str) -> Type[EstimatorT]:
-    estimator_class = locate(class_path)
-
-    if estimator_class is None:
+    else:
         raise GluonTSFatalError(
-            f'Cannot locate estimator with classname "{class_path}".'
+            'Cannot determine the GluonTS estimator classname (missing variable '
+            '"GLUONTS_ESTIMATOR_CLASS").'
         )
 
-    return estimator_class
-
-
-def run() -> None:
-    paths = PathsEnvironment()
-    hyperparameters = get_hyperparameters(paths)
-
-    estimator_path = get_estimator_path(hyperparameters)
-    estimator_class = load_estimator(estimator_path)
-
-    try:
-        from_hyperparameters = getattr(estimator_class, "from_hyperparameters")
-    except AttributeError:
+    Forecaster = locate(forecaster_class)
+    if Forecaster is None:
         raise GluonTSFatalError(
-            'Cannot find static method "from_hyperparameters" for estimator '
-            f'"{estimator_path}"'
+            f'Cannot locate estimator with classname "{forecaster_class}".'
         )
+    return Forecaster
 
-    DefaultShell(paths, hyperparameters, from_hyperparameters).run()
 
+def train(path, forecaster_path) -> None:
+    env = SageMakerEnv(path)
 
-def main() -> None:
-    try:
-        run()
-    except GluonTSException as error:
-        logging.error(error)
-        exit(-1)
+    # forecaster_path = env.hyperparameters.get("forecaster_class")
 
+    Forecaster = get_forecaster(forecaster_path)
+    forecaster = from_hyperparameters(Forecaster, **env.hyperparameters)
```

Comment:
`from_hyperparameters` is only defined on `Estimator` classes, but not on `Predictor`... In general I am not sure whether there is a clean solution of the `Forecaster` union type approach. This will remove the need of a need a noop `train` call but might create other problems elsewhere.

Reply:
The method `from_hyperparameters` on `Estimator` classes is just calling the function I'm using here directly. At the moment it works quite well, and you can use the `Predictor` classes directly.

Reply:
Oh, I think I now get your point. I would say we leave it as it is for now and maybe rethink the `from_hyperparameters` approach in the future when we are more sure about things.

Reply:
We can still remove the need on a per-predictor subclass of a `DummyEstimator` by dispatching on the type of the `predictor_type` after we do `locate(predictor_class)` in the shell. If the type is a subtype of `Estimator`, we call `from_hyperparameters` directly. If it is a subtype of `Predictor`, we can directly call `DummyEstimator` with an extended `config` that also includes the `predictor_type`. I will try to make a concrete proposal in a separate PR.

Reply:
Are you fine with leaving as is for now?